src/subdomains/core/history/mappers/transaction-dto.mapper.ts

```diff
@@ -81,6 +83,7 @@ export class TransactionDtoMapper {
   }
 
   static mapBuyFiatTransaction(buyFiat: BuyFiatExtended): TransactionDto {
+    const referencePrice = buyFiat.inputAmount / buyFiat.inputReferenceAmount;
     const dto: TransactionDto = {
```

Comment:
Calculating the referencePrice here is kind of useless. I would do it the same way as for buy-crypto 😉